MetaLamp/nft-marketplace/src/Plutus/Contracts/Services/Sale/Core.hs

```diff
@@ -42,10 +43,12 @@ type LovelacePrice = Integer
 
 data Sale =
   Sale
-    { saleProtocolToken :: !ThreadToken,
-      salePrice         :: !LovelacePrice,
-      saleValue         :: !Value,
-      saleOwner         :: !Saler
+    { saleProtocolToken   :: !ThreadToken,
+      salePrice           :: !LovelacePrice,
+      saleValue           :: !Value,
+      saleOwner           :: !Saler,
+      marketplaceOperator :: !PubKeyHash,
+      marketplaceSaleFee  :: !Percentage
```

Comment:
Sale is an abstract contract, which should not depend on Marketplace contract. Don't you think it is better to send fees during other transaction which changes Marketplace state?